packages/basti/src/cleanup/security-group-cleaner.ts

```diff
@@ -136,3 +124,50 @@ async function cleanReplicationGroup(
     });
   }
 }
+
+async function cleanElasticacheMemcachedCluster(
+  memcachedCluster: CacheCluster,
+  groupIds: Set<string>
+): Promise<void> {
+  if (
+    memcachedCluster === undefined ||
+    memcachedCluster.SecurityGroups === undefined
+  )
+    return;
+  const parsed = parseMemcachedCacheClusterResponse(memcachedCluster);
+  const cacheSecurityGroupsIds: string[] = filterOut(
+    parsed.securityGroups,
+    groupIds
+  );
+  if (
+    cacheSecurityGroupsIds.length !== memcachedCluster.SecurityGroups.length
+  ) {
+    await modifyElasticacheMemcachedCluster({
+      identifier: memcachedCluster.CacheClusterId,
+      securityGroupIds: cacheSecurityGroupsIds,
+      cachePreviousSecurityGroups: memcachedCluster.SecurityGroups,
+    });
+  }
+}
+
+export async function cleanupElasticacheSecurityGroups(
```

Comment:
I know that it might sound too much, but I'd ask you to slightly change the ordering of the methods again. It's much easier to read the code when it's written top-to-bottom. It's a pretty common convention. Let's raise this function above `cleanReplicationGroup` and `cleanElasticacheMemcachedCluster` since it uses these them. This way, you first see the `cleanupElasticacheSecurityGroups` invocation, then you see that it uses two more functions, you scroll down and can easily find them right below.